### PR TITLE
adds updates for 4_get_articleById

### PR DESCRIPTION
**Pull request by ericnabofa**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -23,3 +23,43 @@ describe('GET /api/topics', () => {
...
+        .expect(200)
+        .then(({body}) => {
+            const actualCreatedAt = body.article.created_at;
+            expect(body.article).toMatchObject({
+                author: 'butter_bridge',
+                title: 'Living in the shadow of a great man',
+                article_id: 1,
+                body: 'I find this existence challenging',
+                topic: 'mitch',
+                created_at: actualCreatedAt,
```
Comment (Harrison365) [inline __tests__/app.test.js:42]: Clever way around this. You could also have expected any string :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -23,3 +23,43 @@ describe('GET /api/topics', () => {
...
+
+    test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
+        return request(app)
+        .get('/api/articles/not-an-article')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+        })
+    });
+});
```
Comment (Harrison365) [inline __tests__/app.test.js]: Nice sad path 👍

File: app/app.controller.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
 
 exports.getAllTopics = (req, res, next) => {
 selectTopics().then((topics) => {
 res.status(200).send({ topics})
 })
 .catch(next)
+}
+
+exports.getArticleByid = (req, res, next) => {
+    const {article_id} = req.params
```
Comment (Harrison365) [inline app/app.controller.js:13]: Nice 👍

File: app/app.model.js [hunk 4]
```diff
@@ -5,4 +5,14 @@ return db.query(`SELECT * FROM topics`)
...
 })
+}
+
+exports.selectArticleById = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({rows}) => {
+        if(!rows.length){
+            return Promise.reject({status: 404, msg: 'article does not exist'})
+        }
+        return rows[0]
```
Comment (Harrison365) [inline app/app.model.js:17]: Nice promise.reject 👍

File: errors.js [hunk 5]
```diff
@@ -1,4 +1,20 @@
...
+      }
+};
+
+exports.handleCustomErrors = (err, req, res, next) => {
+    if (err.status){
+        res.status(err.status).send({ msg: err.msg })
+      } else {
+        next(err)
+      }
+};
```
Comment (Harrison365) [inline errors.js:16]: Nice error handlers 👍

**Comment by Harrison365** [comment]
Great work. Once you have resolved your merge conflicts, you are good to merge ✅ 

Well done 😄

**Comment by ericnabofa** [comment]
thanks

**Pull request merged**